MapboxCoreNavigation/RouteController.swift

```diff
@@ -287,7 +287,7 @@ open class RouteController: NSObject {
      This property contains a `CLLocation` object located along the route line near the most recently received user location. This property is set to `nil` if the route controller is unable to snap the user’s location to the route line for some reason.
      */
     public var location: CLLocation? {
-        guard let location = rawLocation, userIsOnRoute(location) else { return nil }
```

Comment:
Location should not worry about whether the user is on the route. These ideas should be decoupled.

Reply:
Please add these comments to the code.